graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaValidator.kt

```diff
@@ -87,122 +87,4 @@ class FederatedSchemaValidator {
             throw InvalidFederatedSchema(errors)
         }
     }
-
-    // [OK]    @requires references valid fields marked @external
-    // [ERROR] @requires specified on base type
-    // [ERROR] @requires specifies non-existent fields
-    private fun validateRequiresDirective(validatedType: String, validatedField: GraphQLFieldDefinition, fieldMap: Map<String, GraphQLFieldDefinition>, extendedType: Boolean): List<String> {
-        val errors = mutableListOf<String>()
-        if (extendedType) {
-            errors.addAll(validateDirective("$validatedType.${validatedField.name}", "requires", validatedField.directivesByName, fieldMap, extendedType))
-        } else {
-            errors.add("base $validatedType type has fields marked with @requires directive, validatedField=${validatedField.name}")
-        }
-        return errors
-    }
-
-    // [OK]    @provides on base type references valid @external fields on @extend object
-    // [ERROR] @provides on base type references local object fields
-    // [ERROR] @provides on base type references local fields on @extends object
-    // [ERROR] @provides references interface type
-    // [OK]    @provides references list of valid @extend objects
-    // [ERROR] @provides references @external list field
-    // [ERROR] @provides references @external interface field
-    private fun validateProvidesDirective(federatedType: String, field: GraphQLFieldDefinition): List<String> {
```

Comment:
Just moved these functions to their own internal files so we can more easily scope the unit tests